### ISSUE TITLE
Can't delete folder

### ISSUE THREAD

**Issue by aronovgj** [owncloud/core#24819]
### Steps to reproduce
Not sure anymore if this is how it happened. Also not really reproducable:
1. Create folder with files on your PC running ownCloud Client
2. Remove folder on PC before sync is finished

### Expected behaviour
Folder should not be there anymore in cloud and on PC

### Actual behaviour
- Folder is not there on PC (and is not resynct anymore)
- Folder is in the cloud
- If you remove the folder in the browser and reload, the folder reappears

### Server configuration
**Operating system**:
Mint 17.3
**Web server:**
Nginx 1.9.13
**Database:**
mysql 5.5.49
**PHP version:**
5.6.21
**ownCloud version:** (see ownCloud admin page)
8.2.5

**Where did you install ownCloud from:**
https://download.owncloud.org/download/repositories/8.2/Ubuntu_14.04/

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 0.8.2
  - contacts: 0.5.0.0
  - direct_menu: 0.7.1
  - documents: 0.11.1
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - firstrunwizard: 1.1
  - galleryplus: 15.0.0
  - logreader: 1.1.2
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - templateeditor: 0.1
  - updater: 0.6
  - user_external: 0.4
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "ocro13eb0f5c",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "192.168.178.200",
            "www.me.duckdns.org",
            "me.duckdns.org",
            "localhost"
        ],
        "datadirectory": "\/home\/me\/owncloud\/data",
        "overwrite.cli.url": "http:\/\/xxx.xxx.xxx.xxx",
        "overwriteprotocol": "",
        "dbtype": "mysql",
        "version": "8.2.5.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0,
            "dbindex": 0
        },
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "CEST",
        "installed": true,
        "maintenance": false,
        "asset-pipeline.enabled": false,
        "theme": "",
        "loglevel": 1,
        "logfile": "\/var\/log\/owncloud.log",
        "log_rotate_size": 10485760,
        "trashbin_retention_obligation": "auto",
        "mail_from_address": "me",
        "mail_smtpmode": "smtp",
        "mail_domain": "maail.net",
        "mail_smtpsecure": "ssl",
        "mail_smtpport": "465",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "mail.mail.net",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "ldapIgnoreNamingRules": false
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Firefox 46.0.1

**Operating system:**
Windows 7
### Client log
```
|0|indexDir|INST_REMOVE|Up|1463948854|57421636b5cc6|0|00231250ocro13eb0f5c|6|Das Element ist aufgrund vorheriger Fehler nicht synchronisiert: Error downloading https://me.duckdns.org/remote.php/webdav/indexDir - server replied: Forbidden|0|0|0|||INST_NONE|
|0|indexDir/segments|INST_REMOVE|Up|1463948729|1dfa1663e74bd1c091ec389db1588cf6|8|00231252ocro13eb0f5c|0||0|0|0|||INST_NONE|
```

#### ownCloud log (data/owncloud.log)
This is when the file was created. No log for trying to delete.
```
Exception: {"Message":"An exception occurred while executing 'UPDATE `oc_filecache` SET `storage` = ?, `path` = ?, `path_hash` = ? WHERE `fileid` = ?' with params [\"5\", \"files_trashbin\\\/files\\\/indexDir.d1463951852\\\/segments\", \"6db161a82b9951313f8d4e6829fcd1c8\", \"231252\"]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '5-6db161a82b9951313f8d4e6829fcd1c8' for key 'fs_storage_path_hash'","Exception":"Doctrine\\DBAL\\Exception\\UniqueConstraintViolationException","Code":0,"Trace":"#0 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/DBALException.php(116): Doctrine\\DBAL\\Driver\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\DBAL\\Driver\\PDOException))\n#1 \/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Statement.php(172): Doctrine\\DBAL\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\DBAL\\Driver\\PDOMySql\\Driver), Object(Doctrine\\DBAL\\Driver\\PDOException), 'UPDATE `oc_file...', Array)\n#2 \/var\/www\/owncloud\/lib\/private\/db\/statementwrapper.php(73): Doctrine\\DBAL\\Statement->execute(Array)\n#3 \/var\/www\/owncloud\/lib\/private\/db.php(147): OC_DB_StatementWrapper->execute(Array)\n#4 \/var\/www\/owncloud\/lib\/private\/files\/cache\/cache.php(495): OC_DB::executeAudited(Object(OC_DB_StatementWrapper), Array)\n#5 \/var\/www\/owncloud\/lib\/private\/files\/cache\/cache.php(447): OC\\Files\\Cache\\Cache->moveFromCache(Object(OC\\Files\\Cache\\HomeCache), 'files\/indexDir', 'files_trashbin\/...')\n#6 \/var\/www\/owncloud\/lib\/private\/files\/cache\/updater.php(198): OC\\Files\\Cache\\Cache->move('files\/indexDir', 'files_trashbin\/...')\n#7 \/var\/www\/owncloud\/apps\/files_trashbin\/lib\/trashbin.php(225): OC\\Files\\Cache\\Updater->rename('\/files\/indexDir', '\/files_trashbin...')\n#8 \/var\/www\/owncloud\/apps\/files_trashbin\/lib\/storage.php(160): OCA\\Files_Trashbin\\Trashbin::move2trash('indexDir')\n#9 \/var\/www\/owncloud\/apps\/files_trashbin\/lib\/storage.php(106): OCA\\Files_Trashbin\\Storage->doDelete('files\/indexDir', 'rmdir')\n#10 \/var\/www\/owncloud\/lib\/private\/files\/view.php(1024): OCA\\Files_Trashbin\\Storage->rmdir('files\/indexDir')\n#11 \/var\/www\/owncloud\/lib\/private\/files\/view.php(297): OC\\Files\\View->basicOperation('rmdir', '\/indexDir', Array)\n#12 \/var\/www\/owncloud\/lib\/private\/connector\/sabre\/directory.php(246): OC\\Files\\View->rmdir('\/indexDir')\n#13 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Tree.php(179): OC\\Connector\\Sabre\\Directory->delete()\n#14 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(287): Sabre\\DAV\\Tree->delete('indexDir')\n#15 [internal function]: Sabre\\DAV\\CorePlugin->httpDelete(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#16 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#17 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(469): Sabre\\Event\\EventEmitter->emit('method:DELETE', Array)\n#18 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(254): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#19 \/var\/www\/owncloud\/apps\/files\/appinfo\/remote.php(56): Sabre\\DAV\\Server->exec()\n#20 \/var\/www\/owncloud\/remote.php(137): require_once('\/var\/www\/ownclo...')\n#21 {main}","File":"\/var\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/AbstractMySQLDriver.php","Line":66}
```

https://github.com/owncloud/core/issues/22609
Might be related
Similar issues are closed or fixed for earlier oc versions.

**Comment by PVince81** [comment]
@aronovgj would also be good to see the access log so we can see the sequence of commands the desktop client sent to the server. Might help to reproduce this consistently.

CC @icewind1991

**Comment by aronovgj** [comment]
Which access log? Where would I find that? Client side? Server side? I only see the clients owncloudsync.log and the servers owncloud.log.

**Comment by aronovgj** [comment]
I think thats the relevant stuff. 
The owncloud.log error I posted above occurred at 2016-05-22T22:17:33+01:00
Not sure if this is helpful...

```
xxx.xxx.xxx.xxx - me [22/May/2016:22:12:16 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:12:48 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:13:20 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:13:52 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:14:11 +0200] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 500 253 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:14:24 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:14:56 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
84.41.108.114 - - [22/May/2016:22:15:21 +0200] "GET / HTTP/1.1" 301 178 "-" "curl/7.17.1 (mips-unknown-linux-gnu) libcurl/7.17.1 OpenSSL/0.9.8i zlib/1.2.3" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:15:28 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:16:00 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:16:32 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:17:04 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
85.25.148.64 - - [22/May/2016:22:17:09 +0200] "GET /index.html HTTP/1.1" 301 178 "-" "curl/7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.19.1 Basic ECC zlib/1.2.3 libidn/1.18 libssh2/1.4.2" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:17:36 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:18:08 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:18:40 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:19:11 +0200] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 500 253 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:19:12 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:19:44 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:20:16 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:20:32 +0200] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 500 253 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:20:48 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:22:21:00 +0200] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 500 253 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"

```

**Comment by PVince81** [comment]
This is indeed the access log, but there should be an entry with the method "DELETE" which would tell the server to delete something, or other action methods. The log you posted above is only about the client reading the server status but not performing any sync operation.

**Comment by aronovgj** [comment]
Got it, thanks. It was in the access.log, not in the error.log

```
xxx.xxx.xxx.xxx - me [22/May/2016:23:17:33 +0200] "DELETE /remote.php/webdav/indexDir HTTP/1.1" 500 622 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:17:53 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:18:25 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:18:57 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:19:03 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 14929 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:19:11 +0200] "GET /ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 500 253 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:19:29 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:20:01 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:20:06 +0200] "DELETE /remote.php/webdav/indexDir HTTP/1.1" 403 6164 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
xxx.xxx.xxx.xxx - me [22/May/2016:23:20:06 +0200] "DELETE /remote.php/webdav/nonParsableDir HTTP/1.1" 204 0 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"
```

**Comment by PVince81** [comment]
> xxx.xxx.xxx.xxx - me [22/May/2016:23:20:06 +0200] "DELETE /remote.php/webdav/indexDir HTTP/1.1" 403 6164 "-" "Mozilla/5.0 (Windows) mirall/2.2.0 (build 6076)" "-"

Hmm, so you got a 403 Forbidden here.
It looks like it could be https://github.com/owncloud/core/issues/24499.

Does the folder "indexDir" exist in your "data/$user/files" folder ?

**Comment by aronovgj** [comment]
No it doesn't. 

I didn't try this before: in the folder there is still a file which is also not deleted and accessing it gives me an error that the file can't be found on the server. 

It indeed looks like #24499 

Doing a `occ files:scan` would also be my first choice. Just wanted to wait with it for feedback.

Thanks.

**Comment by PVince81** [comment]
Should be fixed in 9.1.1 as per https://github.com/owncloud/core/issues/24499
